File: webapp-net/Ugc/Models/UgcComments.cs
using System.Collections.Generic;
using Sdl.Web.Common.Models;
using Sdl.Web.Modules.Ugc.Data;
using Sdl.Web.Tridion.ContentManager;

namespace Sdl.Web.Modules.Ugc.Models
{
    public class UgcComments : EntityModel
    {
        [SemanticProperty(IgnoreMapping = true)]
        public CmUri Target { get; set; }

        [SemanticProperty(IgnoreMapping = true)]
        public List<UgcComment> Comments { get; set; }
    }
}

File: webapp-java/dxa-module-ugc/src/main/java/com/sdl/dxa/modules/ugc/model/entity/UgcComments.java
package com.sdl.dxa.modules.ugc.model.entity;

import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticMappingIgnore;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;
import com.tridion.util.CMURI;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
public class UgcComments extends AbstractEntityModel {

    /**
     * Target CmUri for comments
     **/
    @SemanticMappingIgnore
    private CMURI target;

    /**
     * List of comments
     **/
    @SemanticMappingIgnore
    private List<UgcComment> comments;
}
